docs/operators/flatten.md

```diff
@@ -0,0 +1,99 @@
+## `flatten` operator
+
+The `flatten` operator flattens a field by moving its children up to the same level as the field.
```

Comment:
My intuition was that this would happen recursively. However, this would introduce a lot complications, so I actually like this version, for now at least.

That said, I think we need to call this out explicitly here, since others may also assume that it is a recursive operation.